packages/insured-bridge-relayer/src/index.ts

```diff
@@ -51,129 +55,155 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
+        const profitabilityCalculator = new ProfitabilityCalculator(
+          logger,
+          filteredL1Whitelist,
+          l1ChainId,
+          config.relayerDiscount
+        );
+
+        const relayer = new Relayer(
+          logger,
+          gasEstimator,
+          l1Client,
+          l2Client,
+          profitabilityCalculator,
+          filteredL1Whitelist,
+          accounts[0],
+          config.whitelistedChainIds,
+          config.l1DeployData,
+          config.l2DeployData,
+          config.l2BlockLookback
+        );
+
+        const canonicalBridgeAdapter = createBridgeAdapter(logger, l1Web3, l2Web3, config.activatedChainIds[0]);
+        if (config.botModes.l1FinalizerEnabled) await canonicalBridgeAdapter.initialize();
+
+        const crossDomainFinalizer = new CrossDomainFinalizer(
+          logger,
+          gasEstimator,
+          l1Client,
+          l2Client,
+          canonicalBridgeAdapter,
+          accounts[0],
+          config.l2DeployData,
+          config.crossDomainFinalizationThreshold
+        );
+
+        return {
+          relayer,
+          crossDomainFinalizer,
+          l2Client,
+          profitabilityCalculator,
+          filteredL1Whitelist,
+        };
+      })
     );
 
     // For all specified whitelisted L1 tokens that this relayer supports, approve the bridge pool to spend them. This
     // method will error if the bot runner has specified a L1 tokens that is not part of the Bridge Admin whitelist.
-    await approveL1Tokens(logger, l1Web3, gasEstimator, accounts[0], config.bridgeAdmin, filteredL1Whitelist);
-
-    // Construct the profitability calculator based on the filteredL1Whitelist and relayerDiscount.
-    const profitabilityCalculator = new ProfitabilityCalculator(
-      logger,
-      filteredL1Whitelist,
-      l1ChainId,
-      config.relayerDiscount
-    );
-
-    const relayer = new Relayer(
-      logger,
-      gasEstimator,
-      l1Client,
-      l2Client,
-      profitabilityCalculator,
-      filteredL1Whitelist,
-      accounts[0],
-      config.whitelistedChainIds,
-      config.l1DeployData,
-      config.l2DeployData,
-      config.l2BlockLookback
-    );
-
-    const canonicalBridgeAdapter = createBridgeAdapter(logger, l1Web3, l2Web3, config.activatedChainIds[0]);
-    if (config.botModes.l1FinalizerEnabled) await canonicalBridgeAdapter.initialize();
-
-    const crossDomainFinalizer = new CrossDomainFinalizer(
-      logger,
-      gasEstimator,
-      l1Client,
-      l2Client,
-      canonicalBridgeAdapter,
-      accounts[0],
-      config.l2DeployData,
-      config.crossDomainFinalizationThreshold
-    );
-
+    const combinedFilteredL1Whitelist = lodash.uniq(relayers.map((relayer) => relayer.filteredL1Whitelist).flat());
+    await approveL1Tokens(logger, l1Web3, gasEstimator, accounts[0], config.bridgeAdmin, combinedFilteredL1Whitelist);
```

Comment:
This just combines the filtered whitelist from all of the L2s in the loop above. It flattens and uniqs the elements before passing into our pre-existing approval function.